apps/scratch/src/hooks/useEmbeddedScratch.ts

```diff
@@ -91,26 +93,33 @@ export const useEmbeddedScratch = (
 
             try {
               // then backup project state
-              const [zip, json] = await Promise.all([
+              const [_, json] = await Promise.all([
                 saveCrtProject(vm).then((blob) => blob.arrayBuffer()),
                 vm.toJSON(),
               ]);
 
               const waitForAssertions = new Promise<{
-                totalTests: number;
-                passedTests: number;
+                passedAssertions: Assertion[];
+                failedAssertions: Assertion[];
               }>((resolve) => {
                 vm.runtime.once(
                   "ASSERTIONS_CHECKED",
-                  (totalTests, passedTests) =>
-                    resolve({ totalTests, passedTests }),
+                  (passedAssertions, failedAssertions) =>
+                    resolve({ passedAssertions, failedAssertions }),
                 );
 
                 // once the project is backed up, run the project
                 vm.greenFlag();
               });
 
-              const { totalTests, passedTests } = await waitForAssertions;
+              const { passedAssertions, failedAssertions } =
+                await waitForAssertions;
+
+              const mapToTest = (assertion: Assertion): Test => ({
+                identifier: `${assertion.targetName} ${assertion.blockId}`,
```

Comment:
@pierluca any opinion for the separator?

Reply:
Sorry I completely missed this message.

What characters are _never_ found in the targetName? `@`, `#`, `:`, `_` or something similar?

Reply:
I think a slash should be fine but I could not find documentation on it :/

Reply:
There is a target id but it is not exported and therefore probably not consistent. As for the block id, `jEk@4|i[#Fk?(8x)AV` is a valid id

Reply:
Although there do not seem to be spaces or slashes meaning we can search from the end of the identifier finding the **last** occurrence and split there..

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/lastIndexOf

Reply:
Found it: https://github.com/scratchfoundation/scratch-vm/blob/c563ee008b9d0e0ebe871b54e8dea4afad568717/src/util/uid.js#L11-L12. A slash is not find, a dollar sign is however!